src/loop.c

```diff
@@ -340,11 +444,49 @@ int mosquitto_main_loop(struct mosquitto_db *db, mosq_sock_t *listensock, int li
 
 #ifndef WIN32
 		sigprocmask(SIG_SETMASK, &sigblock, &origsig);
+#ifdef WITH_EPOLL
+		fdcount = epoll_wait(db->epollfd, events, MAX_EVENTS+1, 100);
```

Comment:
Why there is a +1 on MAX_EVENTS ? To me it looks like it may cause a memory overflow (since epoll_wait may return 1001 entry and events can only hold 1000)